src/domain/coreConnectorAgg.ts

```diff
@@ -25,163 +25,303 @@
         };
-        const transferRes = await this.fineractClient.transfer({
+        const account = await this.fineractClient.getSavingsAccount(res.accountId);
+        if (account.statusCode != 200) {
+            throw new FineractGetAccountWithIdError('Failed to retrieve destination account from fineract', 'MFCC Agg');
+        } else if (!account.data.status.active) {
+            throw new AccountVerificationError('Funds Destination Account is not active in Fineract', 'MFCC Agg');
+        } else if (account.data.subStatus.blockCredit) {
+            throw new FineractAccountDebitOrCreditBlockedError('Account blocked from credit', 'MFCC Agg');
+        }
+        await this.fineractClient.receiveTransfer({
             accountId: res.accountId as number,
-            transaction: transaction
+            transaction: transaction,
         });
-        if(!transferRes){
-            return;
-        }
-        const transferResponse : TtransferResponse = {
-            completedTimestamp : new Date().toJSON(),
+        return {
+            completedTimestamp: new Date().toJSON(),
             homeTransactionId: transfer.transferId,
-            transferState: "COMMITTED"
+            transferState: 'COMMITTED',
+        };
+    }
+
+    async sendTransfer(transfer: TFineractOutboundTransferRequest): Promise<TFineractOutboundTransferResponse> {
+        this.logger.info(`Transfer from fineract account with ID${transfer.from.fineractAccountId}`);
+        const account = await this.fineractClient.getSavingsAccount(transfer.from.fineractAccountId);
+        if (account.statusCode != 200) {
+            throw new FineractGetAccountWithIdError('Failed to retrieve source account from fineract', 'MFCC Agg');
+        } else if (!account.data.status.active) {
+            throw new AccountVerificationError('Funds Source Account is not active in Fineract', 'MFCC Agg');
+        } else if (account.data.subStatus.blockCredit || account.data.subStatus.blockDebit) {
+            throw new FineractAccountDebitOrCreditBlockedError('Account blocked from credit or debit', 'MFCC Agg');
+        }
+        const sdkOutboundTransfer: TSDKOutboundTransferRequest = this.getSDKTransferRequest(transfer);
+        const transferRes = await this.sdkClient.initiateTransfer(sdkOutboundTransfer);
+        if (transferRes.statusCode != 200) {
+            throw new SDKClientInitiateTransferError(
+                `Transfer initiation failed with status code ${transferRes.statusCode}`,
+                'MFCC Agg',
+            );
+        } else if (!transferRes.data.quoteResponse) {
+            throw new SDKNoQuoteReturnedError('Quote response is not defined', 'MFCC Agg');
+        }
+        const totalFineractFee = await this.fineractClient.calculateWithdrawQuote({
+            amount: this.getAmountSum([
+                parseFloat(transferRes.data.amount),
+                parseFloat(transferRes.data.quoteResponse.body.payeeFspFee?.amount as string),
```

Comment:
you can't just so `as string` and get a number.  `as string` works only for Typescript during compilation, but during runtime it might be a problem. First of all, `quoteResponse` is optional, so you might get `Uncaught TypeError: Cannot read properties of undefined`. Next, if `quoteResponse` exists, but no `payeeFspFee`, parseFloat will return `NaN`